### ISSUE TITLE
Establish MPFB on the extensions platform

### ISSUE THREAD

**Issue by Vidyut** [makehumancommunity/mpfb2#190]
Checked out Blender 4.2 alpha - it introduces repositories for plugins so you can download and install plugins from inside blender - like wordpress, etc.

Should probably  be simple enough to get MPFB there? (and any other plugins appropriate) It should increase adoption of the plugin considerably as well as make frequent updates from the nightlies easier. I don't know what is involved, but seems like a simple upload situation.

**Comment by joepal1976** [comment]
There's a few parts of this:

- Making sure everything works with 4.2. I've done some initial tests, and so far I've only seen minor glitches
- Converting MPFB from a "legacy" addon module format to the new namespaced module format. This is so far a bit unknown how much work it is. On paper it shouldn't be much, but that's for small one-file addons.
- Ensuring MPFB follows the guidelines for the extension platform. This is #86 
- Updating the build pipelines to allow for automatically updating the addon on the extensions platform

The problem is that you can't combine having a) compatibility with the extensions platform and b) compatibility with blender 4.1 and earlier. So either two parallel builds will have to be maintained or support for blender 4.1 and earlier has to be dropped.

The Blender devs sure do everything humanly possible to force addon developers onto the extension platform. This includes calling every single today existing addon "legacy" and hiding the ability to install those addons in obscure corners of the UI...

**Comment by Vidyut** [comment]
Ah ok. So it won't simply be a matter of setting up builds for two platforms then... I was thinking that the one on the site would be as usual and an additional one would go there. Well, it was a thought. Thanks for accepting all these crazy ideas anyway.

**Comment by joepal1976** [comment]
This is blocked by #208, but is now planned for a second beta early fall.

**Comment by joepal1976** [comment]
With #208 done, I have now been able to refactor MPFB from legacy addon to extension. This works technically as far as I can see. The extension version is available in the "as_extension" branch: https://github.com/makehumancommunity/mpfb2/tree/as_extension

Blender gets very confused if an addon is suddenly an extension, so before testing this it probably makes sense to completely remove any currently installed legacy addon version of MPFB.

Still need to update build pipelines and figure out which guidelines are relevant before posting on the extensions platform.

**Comment by Vidyut** [comment]
>Blender gets very confused if an addon is suddenly an extension, so before testing this it probably makes sense to completely remove any currently installed legacy addon version of MPFB.

I didn't read this! I think my issues with my old files looking glassy is probably because I didn't uninstall enough. Though I still haven't go time to figure out loading the data in 4.3. It is buggy with a few other things as well, so I'm not sure if this is blender 4.3 or some other addon creating trouble... On one hand I like the repository system with extensions - needing a new name because of the structure also makes sense, but the whole "legacy" thing and hiding the install button like it is a shameful secret... I think it might be a bit "political" - paid addons can't be on the free repository, so the users are being encouraged to use free ones by making the paid ones a bit inconvenient/worrying. Unfortunately, this also applies to loads of free addons, like ours.

The great part of Blender is that it is evolving so rapidly, with new possibilities opening up every moment. The downside is that by the time I get everything working in one version, it is time to update and they don't seem to care much about backwards compatibility, so something or the other will freak out. And now MPFB2 must also follow this curve... I'm thinking of stopping updating once I figure 4.3 out and just figuring out one thing and using it instead of a permanent figuring out curve.

**Comment by joepal1976** [comment]
The working theory is to submit a first version of MPFB for review on the extension platform during the coming week. Then there's no way to predict what changes are requested by the blender guys.

It might be overly optimistic, but I hope to have MPFB released as stable and available for download from the extension platform this side of new year.

**Comment by Vidyut** [comment]
When you write a description, be sure to use "human generator and asset management system" in it. We'll probably welcome the new year with a crowd of new adopters. :D 

I'm seriously surprised why this plugin isn't talked about more. I like the look of these characters far more than other human generators. Perhaps it is a topology thing for advanced users or something, but for my skill level (and laptop's willingness), this is so brilliant, it feels like everybody should be using it. Just the idea of the same character being easily generated with different rigs for different purposes would do the trick, let alone the extensive bells and whistles.

(Or maybe it is just sponsorships by paid addons vs a free addon that's been around forever...)

**Comment by joepal1976** [comment]
Well, here goes... The addon has been submitted for approval at the extensions platform: https://extensions.blender.org/approval-queue/mpfb/

Seeming as there are addons which have been submitted two months ago and which have not yet been reviewed, maybe new year is unlikely. 

As a side note: I have no idea what happens if one tries to install the extension from the extensions platform when already having a nightly build or a beta installed. It's probably best to uninstall MPFB before trying the version from the platform.

**Comment by Vidyut** [comment]
Merry Christmas! Huge job completed. Well done!

**Comment by joepal1976** [comment]
@Vidyut Merry christmas to you too. :-)

And thank you for the enouragment.

**Comment by joepal1976** [comment]
The first batch of feedback has arrived. I'll handle each batch in a separate issue to keep track of the changes.

The first batch is in #231

**Comment by Vidyut** [comment]
The default thumbnail on the extension platform listing is a white face. Change it to one with realistic skin to reassure users that this is indeed what they were praying for.

**Comment by joepal1976** [comment]
> The default thumbnail on the extension platform listing is a white face. Change it to one with realistic skin to reassure users that this is indeed what they were praying for.

Yeah, that's a good idea. 

Anyway, closing this issue since the addon is now approved on the extension platform.
